python/arcticdb/version_store/library.py

```diff
@@ -1383,9 +1383,12 @@ def finalize_staged_data(
         2024-01-03    3
         2024-01-04    4
         """
+        if not (mode is None or isinstance(mode, StagedDataFinalizeMethod) or mode == "WRITE" or mode == "APPEND"):
```

Comment:
Should validate that the enum value is either `WRITE` or `APPEND` as protection against someone changing the enum and us silently treating it as `WRITE`